tests/backup/backup_upgrade_test.go

```diff
@@ -204,3 +205,366 @@ var _ = Describe("{StorkUpgradeWithBackup}", func() {
+				err = backupSuccessCheck(secondScheduleBackupName.(string), orgID, maxWaitPeriodForBackupCompletionInMinutes*time.Minute, 30*time.Second, ctx)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying success of second schedule backup [%s] with schedule [%s]", secondScheduleBackupName.(string), singleNSScheduleNames[index]))
+				log.Infof("Second schedule backup name: [%s]", secondScheduleBackupName.(string))
+				// Default Restore
+				restoreName := fmt.Sprintf("%s-%s", "test-restore-single-ns", RandomString(4))
+				namespaceMapping := make(map[string]string, 0)
+				storageClassMapping := make(map[string]string, 0)
+				log.InfoD("Restoring backup [%s] in cluster [%s] with restore [%s]", secondScheduleBackupName.(string), destinationClusterName, restoreName)
+				err = CreateRestore(restoreName, secondScheduleBackupName.(string), namespaceMapping, destinationClusterName, orgID, ctx, storageClassMapping)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying restoration [%s] of backup [%s] in cluster [%s]", restoreName, secondScheduleBackupName.(string), restoreName))
+				// Custom Restore
+				customRestoreName := fmt.Sprintf("%s-%s", "test-custom-restore-single-ns", RandomString(4))
+				customNamespaceMapping := map[string]string{namespace: "custom" + namespace}
+				customStorageClassMapping := make(map[string]string, 0)
+				log.InfoD("Restoring backup [%s] in cluster [%s] with restore [%s] and namespace mapping %v", secondScheduleBackupName.(string), destinationClusterName, customRestoreName, customNamespaceMapping)
+				err = CreateRestore(customRestoreName, secondScheduleBackupName.(string), customNamespaceMapping, destinationClusterName, orgID, ctx, customStorageClassMapping)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying restoration [%s] of backup [%s] in cluster [%s]", customRestoreName, secondScheduleBackupName.(string), customRestoreName))
+				// Restore while replacing existing resources
+				replaceRestoreName := fmt.Sprintf("%s-%s", "test-replace-restore-single-ns", RandomString(4))
+				log.InfoD("Restoring backup [%s] while replacing existing resources in cluster [%s] with restore [%s]", secondScheduleBackupName.(string), destinationClusterName, replaceRestoreName)
+				replaceNamespaceMapping := make(map[string]string, 0)
+				replaceStorageClassMapping := make(map[string]string, 0)
+				err = CreateRestoreWithReplacePolicy(replaceRestoreName, secondScheduleBackupName.(string), replaceNamespaceMapping, destinationClusterName, orgID, ctx, replaceStorageClassMapping, ReplacePolicy_Delete)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying restoration [%s] of backup [%s] in cluster [%s]", replaceRestoreName, secondScheduleBackupName.(string), replaceRestoreName))
+				restoreNames = SafeAppend(&mutex, restoreNames, restoreName, customRestoreName, replaceRestoreName).([]string)
+			}
```

Comment:
use separate mutex for `restoreNames `, as lock should "belong" to the object that it is protecting.